source/iface/properties.rst

```diff
@@ -2,20 +2,191 @@
+class and the SYCL property classes is provided below.
+
+::
+
+  namespace sycl {
+
+  template <typename Property> struct is_property;
+
+  template <typename Property>
+  inline constexpr bool is_property_v = is_property<Property>::value;
+
+  template <typename Property, typename SyclObject> struct is_property_of;
+
+  template <typename Property, typename SyclObject>
+  inline constexpr bool is_property_of_v =
+      is_property_of<Property, SyclObject>::value;
+
+  class T {
+    ...
+
+        template <typename Property>
+        bool has_property() const noexcept;
+
+    template <typename Property> Property get_property() const;
+
+    ...
+  };
+
+  class property_list {
+   public:
+    template <typename... Properties> property_list(Properties... props);
+  };
+  } // namespace sycl
+
+(constructor)
+==============
+
+::
+
+  template <typename... PropertyN> property_list(PropertyN... props)
+
+Available only when: ``is_property<property>::value`` evaluates to
+``true`` where ``property`` is each property in ``PropertyN``.
+
+Construct a SYCL ``property_list`` with zero or more properties.
+
+.. _traits_for_properties :
+
+=====================
+Traits for properties
+=====================
+
+``is_property``
 ===============
- property_list
-===============
 
 ::
 
-   class property_list;
+  template <typename Property> struct is_property
+
+An explicit specialization of ``is_property`` that inherits from
+``std::true_type`` must be provided for each property, where
+``Property`` is the class defining the property. This includes both
+standard properties described in this specification and any additional
+non-standard properties defined by an implementation. All other
+specializations of ``is_property`` must inherit from ``std::false_type``.
+
+``is_property_v``
+=================
+
+::
+
+  template <typename Property> inline constexpr bool is_property_v;
+
+Variable containing value of ``is_property<Property>``.
+
+``is_property_of``
+==================
+
+::
+
+  template <typename Property, SyclObject> struct is_property_of
 
-property_list
-=============
+An explicit specialization of ``is_property_of`` that inherits from
+``std::true_type`` must be provided for each property that can be used
+in constructing a given SYCL class, where ``Property`` is the class defining
+the property and ``SyclObject`` is the SYCL class. This includes both standard
+properties described in this specification and any additional non-standard
+properties defined by an implementation. All other specializations of
+``is_property_of`` must inherit from ``std::false_type``.
+
+``is_property_of_v``
+====================
 
 ::
 
-   template <typename... propertyTN>
-   property_list(propertyTN... props);
+  template <typename Property, SyclObject> inline constexpr bool is_property_of_v;
+
+Variable containing value of ``is_property_of<Property, SyclObject>``.
+
+================
+Member functions
+================
+
+``has_property``
```

Comment:
better to make it and functions below:
```
``has_property``
----------------
```

Reply:
from my perspective is better to be visible in list from begging

Reply:
I think we should preserve hierarchy... There is an option for the contents box to make the depth bigger (we can ask Khronos WG if we can change it from 1 to 2, so contents on depth 2 will be shown)